test/populate.spec.js

```diff
@@ -2945,4 +2954,184 @@ describe('Model populate tests', function () {
       })
     })
   })
+
+  describe('regression', function (dene) {
+    it('should populate refs in array with schema having prefix #129', function (done) {
+      const addressSchema = lounge.schema({
+        street: String,
+        city: String,
+        country: String
+      })
+
+      const Address = lounge.model('Address_129_1', addressSchema)
+
+      const blogPostSchema = lounge.schema(
+        {
+          title: String,
+          body: String
+        },
+        {
+          keyPrefix: 'blog::'
+        }
+      )
+
+      const BlogPost = lounge.model('BlogPost_129_1', blogPostSchema)
+
+      const userSchema = lounge.schema({
+        name: String,
+        address: Address,
+        posts: [BlogPost]
+      })
+
+      const User = lounge.model('User_129_1', userSchema)
+
+      const post = new BlogPost({
+        title: 'Foo',
+        body: 'Lorem ipsum'
+      })
+
+      let user = new User({
+        name: 'Bob Smith',
+        posts: [post],
+        address: new Address({
+          street: '123 Fake Street',
+          city: 'Springfield',
+          country: 'USA'
+        })
+      })
+
+      user.posts.push(
+        new BlogPost({
+          title: 'Post 2',
+          body: 'Some more text!'
+        })
+      )
+
+      user.save((e_, savedUser) => {
+        User.findById(savedUser.id, { populate: true }, (err, doc) => {
+          expect(err).to.not.be.ok
+          expect(doc).to.be.ok
+
+          expect(doc.id).to.equal(savedUser.id)
+          expect(doc.name).to.equal('Bob Smith')
+
+          expect(doc.address).to.be.ok
+          expect(doc.address).to.be.an.instanceof(Address)
+          expect(doc.address.id).to.be.ok
+          expect(doc.address.street).to.equal('123 Fake Street')
+          expect(doc.address.city).to.equal('Springfield')
+          expect(doc.address.country).to.equal('USA')
+
+          expect(doc.posts).to.be.ok
+          expect(doc.posts).to.be.an.instanceof(Array)
+          expect(doc.posts[0]).to.be.an.instanceof(BlogPost)
+          expect(doc.posts[0].id).to.be.ok
+          expect(doc.posts[0].title).to.be.ok
+          expect(doc.posts[0].title).to.be.a('string')
+          expect(doc.posts[0].title).to.equal('Foo')
+          expect(doc.posts[1]).to.be.an.instanceof(BlogPost)
+          expect(doc.posts[1].id).to.be.ok
+          expect(doc.posts[1].title).to.be.ok
+          expect(doc.posts[1].title).to.be.a('string')
+          expect(doc.posts[1].title).to.equal('Post 2')
+
+          done()
+        })
+      })
+    })
+
+    it('should populate refs in array with schema having prefix #129', function (done) {
```

Comment:
Are these two tests different in any way? They have the same description.

Reply:
Yes in 2nd scenario all the schemas have a key prefix defined I'll upgrade the description once I have access to internet and git.